pywhy_graphs/networkx/algorithms/causal/m_separation.py

```diff
@@ -21,24 +22,9 @@ def m_separated(
     """Check m-separation among 'x' and 'y' given 'z' in mixed-edge causal graph G, which may
     contain directed, bidirected, and undirected edges.
 
-    This algorithm adapts the linear time algorithm presented in [1]_ currently implemented in
-    `networkx.algorithms.d_separation` to work for mixed-edge causal graphs, using m-separation
-    logic detailed in [2]_.
-
-    This algorithm works by retaining select edges in each of the directed, bidirected, and
-    undirected edge subgraphs (if supplied). Then, an undirected graph is created from
-    the union of allsuch retained edges (without direction information), and then
-    m-separation of x and y givenz is determined if x is disconnected from y in this graph.
-
-    In the directed edge subgraph, nodes and associated edges are removed if they are childless
-    and not in x | y | z; this process is repeated until no such nodes remain. Then, outgoing
-    edges from z are removed. The remaining edges are retained.
+    This implements the m-separation algorithm TESTSEP presented in [1]_, with additional
+    checks to ensure that it works for non-ancestral mixed graphs (e.g. ADMGs).
```

Comment:
Just for the networkx sake, in case we can get this function into their package, perhaps we can add a short description of how the algorithm works and perhaps the big-O runtime.

Reply:
Added!